go/arrow/flight/server.go

```diff
@@ -64,6 +63,24 @@ func RegisterFlightServiceServer(s grpc.ServiceRegistrar, srv FlightServer) {
 	flight.RegisterFlightServiceServer(s, srv)
 }
 
+// From https://github.com/grpc/grpc-go/blob/4c776ec01572d55249df309251900554b46adb41/reflection/serverreflection.go#L69-L83
+// See "google.golang.org/grpc/reflection" 's reflection.ServiceInfoProvider
+// ServiceInfoProvider is an interface used to retrieve metadata about the
+// services to expose.
+//
+// The reflection service is only interested in the service names, but the
+// signature is this way so that *grpc.Server implements it. So it is okay
+// for a custom implementation to return zero values for the
+// grpc.ServiceInfo values in the map.
+//
+// Experimental
+//
+// Notice: This type is EXPERIMENTAL and may be changed or removed in a
+// later release.
+type serviceInfoProvider interface {
+	GetServiceInfo() map[string]grpc.ServiceInfo
+}
```

Comment:
Can you add a small comment about the fact that we're inlining this in order to provide compatibility with grpc < 1.45?

Reply:
Yep sure thing. Any thoughts about the comment copied verbatim from https://github.com/grpc/grpc-go/blob/4c776ec01572d55249df309251900554b46adb41/reflection/serverreflection.go#L69-L83 ?

Reply:
I think it's fine copying the comment verbatim for now.